crates/swc_ecma_transforms_module/src/system_js.rs

```diff
@@ -0,0 +1,1189 @@
+    pub allow_top_level_this: bool,
+}
+
+struct SystemJs<R>
+where
+    R: ImportResolver,
+{
+    root_mark: Mark,
+    resolver: Option<(R, FileName)>,
+    config: Config,
+
+    declare_var_idents: Vec<Ident>,
+    export_map: AHashMap<Id, Vec<JsWord>>,
+    export_names: Vec<JsWord>,
+    export_values: Vec<Box<Expr>>,
+    tla: bool,
+    enter_async_fn: bool,
+    enter_fn: bool,
+    root_fn_decl_idents: Vec<Ident>,
+    module_item_meta_list: Vec<ModuleItemMeta>,
+    import_ident_names: Vec<JsWord>,
+}
+
+pub fn system_js(root_mark: Mark, config: Config) -> impl Fold {
+    SystemJs {
+        root_mark,
+        resolver: None::<(NoopImportResolver, _)>,
+        config,
+
+        declare_var_idents: vec![],
+        export_map: Default::default(),
+        export_names: vec![],
+        export_values: vec![],
+        tla: false,
+        enter_async_fn: false,
+        enter_fn: false,
+        root_fn_decl_idents: vec![],
+        module_item_meta_list: vec![],
+        import_ident_names: vec![],
+    }
+}
+
+pub fn system_js_with_resolver<R>(
+    resolver: R,
+    base: FileName,
+    root_mark: Mark,
+    config: Config,
+) -> impl Fold
+where
+    R: ImportResolver,
+{
+    SystemJs {
+        root_mark,
+        resolver: Some((resolver, base)),
+        config,
+
+        declare_var_idents: vec![],
+        export_map: Default::default(),
+        export_names: vec![],
+        export_values: vec![],
+        tla: false,
+        enter_async_fn: false,
+        enter_fn: false,
+        root_fn_decl_idents: vec![],
+        module_item_meta_list: vec![],
+        import_ident_names: vec![],
+    }
+}
+
+struct ModuleItemMeta {
+    export_names: Vec<JsWord>,
+    export_values: Vec<Box<Expr>>,
+    has_export_all: bool,
+    src: JsWord,
+    setter_fn_stmts: Vec<Stmt>,
+}
+
+impl<R> SystemJs<R>
+where
+    R: ImportResolver,
+{
+    fn export_call(&self, name: JsWord, span: Span, expr: ExprOrSpread) -> CallExpr {
+        CallExpr {
+            span,
+            callee: Callee::Expr(Box::new(Expr::Ident(quote_ident!("_export")))),
+            args: vec![Expr::Lit(Lit::Str(quote_str!(name))).as_arg(), expr],
+            type_args: None,
+        }
+    }
+
+    fn fold_module_name_ident(&mut self, ident: Ident) -> Expr {
+        if ident.sym.eq("__moduleName") && ident.span.ctxt().outer().eq(&self.root_mark) {
+            return Expr::Member(MemberExpr {
+                span: DUMMY_SP,
+                obj: Box::new(Expr::Ident(quote_ident!("_context"))),
+                prop: MemberProp::Ident(quote_ident!("id")),
+            });
+        }
+        Expr::Ident(ident)
+    }
+
+    fn replace_assign_expr(&mut self, assign_expr: AssignExpr) -> Expr {
+        match &assign_expr.left {
+            PatOrExpr::Expr(pat_or_expr) => match &**pat_or_expr {
+                Expr::Ident(ident) => {
+                    for (k, v) in self.export_map.iter() {
+                        if ident.to_id().eq(k) {
```

Comment:
Please use `==` instead of `.eq`. You may need to reference or dereference operands.

Reply:
I think.`ident.to_id() == *k` will work